### DISCUSSION TITLE
Shared notes in read mode only (copying not able)

### DISCUSSION THREAD

**Discussion by GuanFuling** [alangrainger/share-note#106] [Vote for upcoming features]
Hi ! Thank you for your very useful plugin ! I would like to share some notes but in read mode only, neither modifications or copy are able. So i wonder if it's possible to add this option in any way with share note. I am not a programmer at all, i tried to add a html script (wich works in html file) to prevent any selection and copy directly in the obsidian file before sharing it, but it didn't work. Until now my best solution is to share a html file through google frive, people have to download it. So your plugin would be very good with this feature. Thanks a lot again to you and all the community efforts that lead to the excellent Obsidian software !

**Comment by alangrainger** [comment]
Hi @GuanFuling , currently the notes are shared read only. While it's always possible to edit an HTML page by using the inspector, changes will not be saved. 

It's also not really possible to stop someone copying. Once the data has arrived in the recipient's computer, they already have it. Where there's a will there's a way :)

**Reply by GuanFuling** [reply]
Thank you for your fast replying ! Yes there is always a way to dodge the limitations, but i meant to disable the selection of the text and the right-click of the mouse, or the command ctrl+c/+v. I did that for some html pages, so i wonder if it was possible to disable that in share note.

**Reply by alangrainger** [reply]
It's possible to disable text selection etc, but that's not in scope to be added for this particular project sorry.

**Reply by GuanFuling** [reply]
Ok thank you, no don't be sorry. Even if i don't need this plugin for the moment, I tried different similar plugin and i have to tell you that yours is the most convenient and easy to use i've found so far.

**Comment by NA0341** [comment]
@GuanFuling
**About sharing Sensible Information.**

An Easy and one of the "safest" Methods I know is to Print/Export the Note to PDF and Convert the PDF to PostScript.

On GNU/Linux, you can do this easily with the `pdf2ps` - Command (preinstalled on most systems):

`pdf2ps [options] input.pdf [output.ps]`

For other Systems, you can lookup a good way to convert by searching for: `[OS-name] convert PDF to PS`
(I know this is self - explanatory ~ but I felt like writing it :D)

---

**Edit:** I don't know if that's a good idea, but one could render the PostScript into the WebView of the Site.

OR Render the Note's Page and Convert it into Images (vector or raster) which contain no selectable text.

I think this would be doable fairly easily for small notes with SVG ~ but as their size and complexity grows, issues might arise ~ but not necessarily since very big SVG's like the `Linux Distribution's Timeline` render just fine in all up-to-date WebBrowsers.

In practice there could be an additional Function in the Plugin (like `Create Copy-Safe Link`) which in turn creates a Link with Different Syntax that indicates that this Site is Copy-Protected.

Additional Suggestions:
+ Add a transparent layer over the entire Canvas to **block access** to the Image.
+ Embed the SVG directly into the main HTML of the site. That way it gets harder to extract it ~ and the site loads much faster because there's only one resource to fetch.
+ Make the embedded SVG encrypted. (This would make it Really Hard to get the actual image ~ but well ~ Screenshotting the Page is not that difficult.)

It'd be of course still possible to extract the SVG, convert it to raster and apply OCR to it. But the goal of making stealing of the content has gotten much more difficult this way.

---
**IMPORTANT NOTE:**

Converting Notes to SVG might even be a way to make Site Loading and even Rendering faster. Because there's no CSS, Js, etc. to load. (I read about the Issue with Too Much Data due to the Obsidian Theme).

If the Page get's rendered as an SVG (this time with regular text so it's selectable) and served embedded into the main HTML File ~ this might bring Benefits. Especially because there are high-quality Existing Libraries & Applications which can do that.

Maybe worth a try @alangrainger?
